torchbench.py

```diff
@@ -730,18 +739,29 @@ def main():
 
     experiment = functools.partial(experiment, args, model_iter_fn)
 
+    if args.speedup_fx2trt:
+        cos_similarity = True
+    else:
+        cos_similarity = False
+
     if output_filename:
         output_filename = os.path.join(torchdynamo.config.base_dir, output_filename)
 
     if args.minimum_call_count:
         torchdynamo.config.minimum_call_count = args.minimum_call_count
-
     if args.only:
         for device in args.devices:
             try:
                 device, name, model, example_inputs = load_model(
                     device, args.only, args.training, args.check_accuracy
                 )
+
+                if args.speedup_fx2trt:
+                    # cast model and inputs to fp16
```

Comment:
Have you tried using fx2trt with `explicit_precision=True`?  When I tested things this it caused the types to stay float32 and pass the accuracy tests.  I'd be curious to see how much of the speedup is from reduced precision.

Is that mode still supported?  Perhaps we could have a `--speedup-fx2trt-fp16` and an `--speedup-fx2trt` allowing us to compare them both.

Nit: refactor the "convert to fp16" logic into its own function.

Reply:
I am not sure how is that works but according to TRT(EXPLICIT_PRECISION : Specify that the network contains explicit quantization and dequantization scale layers.) It's something related with int8. 
I created a `--speedup-fx2trt-fp16` and an `--speedup-fx2trt` and compare the results on some vision models. Fp32 can not pass the current 1e-4 threshold. (In backend.py:342, I still set explicit_precision=True to see if it works ). Here are the results I got:
Alexnet:
Fp16
tensor([ 0.0450, -1.0811, -1.4688,  ..., -1.4307, -0.8989,  1.1025],
       device='cuda:0')
tensor([ 0.0420, -1.0791, -1.4688,  ..., -1.4307, -0.8994,  1.1055],
       device='cuda:0')

Fp32
tensor([ 0.0459, -1.0811, -1.4692,  ..., -1.4308, -0.8991,  1.1039],
       device='cuda:0')
tensor([ 0.0457, -1.0810, -1.4689,  ..., -1.4307, -0.8994,  1.1040],
       device='cuda:0')

Resnet18:
Fp16:
tensor([1.0850, 2.3184, 2.3320,  ..., 0.3833, 0.4138, 1.2676], device='cuda:0')
tensor([1.0918, 2.3145, 2.3281,  ..., 0.3867, 0.4097, 1.2656], device='cuda:0')

Fp32:
tensor([1.0875, 2.3192, 2.3329,  ..., 0.3841, 0.4151, 1.2665], device='cuda:0')
tensor([1.0907, 2.3191, 2.3355,  ..., 0.3846, 0.4158, 1.2661], device='cuda:0')

Resnet50:
Fp16:
tensor([-0.1942, -0.5420,  1.2812,  ..., -1.0527, -1.2080,  2.0488],
       device='cuda:0')
tensor([-0.1941, -0.5420,  1.2793,  ..., -1.0518, -1.2080,  2.0547],
       device='cuda:0')

Fp32:
tensor([-0.1956, -0.5414,  1.2803,  ..., -1.0534, -1.2087,  2.0509],
       device='cuda:0')
tensor([-0.1929, -0.5416,  1.2817,  ..., -1.0531, -1.2101,  2.0512],
       device='cuda:0')
 
mobilenet_v2
Fp16:
tensor([-1.8945,  0.1934,  0.8633,  ..., -1.1885,  0.0638,  1.7451],
       device='cuda:0')
tensor([-1.8770,  0.2068,  0.8896,  ..., -1.1963,  0.0406,  1.7578],
       device='cuda:0')

Fp32:
tensor([-1.8944,  0.1888,  0.8605,  ..., -1.1921,  0.0652,  1.7422],
       device='cuda:0')
tensor([-1.8936,  0.1904,  0.8655,  ..., -1.2234,  0.0559,  1.7499],
       device='cuda:0')

mnasnet1_0
Fp16:
tensor([-3.9883, -1.2520, -3.3340,  ..., -0.8306,  0.1199,  1.8096],
       device='cuda:0')
tensor([-4.0195, -1.2705, -3.3828,  ..., -0.8257,  0.1372,  1.8242],
       device='cuda:0')
Fp32:
tensor([-3.9908, -1.2514, -3.3324,  ..., -0.8314,  0.1192,  1.8083],
       device='cuda:0')
tensor([-3.9767, -1.2598, -3.3286,  ..., -0.8319,  0.1158,  1.7996],
       device='cuda:0')